ethpm/deployments.py

```diff
@@ -1,5 +1,50 @@
-class Deployments:
+from ethpm.exceptions import ValidationError
+
+from ethpm.utils.contract import (
+    validate_contract_name,
+)
+
 
-    def __init__(self, deployment_data, w3):
+class Deployments:
+    """
+    Deployment object to access instances of
+    deployed contracts belonging to a package.
+    """
+    def __init__(self, deployment_data, contract_types, w3):
         self.deployment_data = deployment_data
+        self.contract_types = contract_types
         self.w3 = w3
+
+    def __getitem__(self, key):
+        self._validate_name_and_references(key)
+        return self.deployment_data[key]
+
+    def __contains__(self, key):
+        return key in self.deployment_data
+
+    def items(self):
+        for name in self.deployment_data:
+            self._validate_name_and_references(name)
+        return self.deployment_data.items()
+
+    def values(self):
+        for name in self.deployment_data:
+            self._validate_name_and_references(name)
+        return list(self.deployment_data.values())
+
+    def get_contract_instance(self, contract_name):
+        """
+        Fetches a contract instance belonging to deployment
+        after validating contract name.
+        """
+        self._validate_name_and_references(contract_name)
+        raise NotImplementedError("All checks passed, but get_contract_instance API not complete.")
+
+    def _validate_name_and_references(self, name):
```

Comment:
should this be pulled out into a util?

Reply:
No, but what I would do this.

* Add a new `.get(contract_name)` method and inline this validation in that method.
* Rewrite *all* of the `values()`, `items()`, and `__getitem__()` methods to use `self.get(contract_name)` internally so that there is a single point where validation happens and all other APIs make use of that single point of entry.